xdsl/dialects/transform.py

```diff
@@ -614,14 +614,19 @@ def __init__(
                 [
                     AnyOpType()
                     for _ in range(
-                        len(static_sizes.as_tuple())
-                        if isinstance(static_sizes, DenseArrayBase)
-                        else 0
+                        self.amount_of_non_zero_sizes(static_sizes, dynamic_sizes)
```

Comment:
I ask this because our usual convention would make this `num_non_zero_sizes)`, but if MLIR calls it this way then we can follow their convention, as long as we add a link to their implementation.

Reply:
BTW this implementation is a little surprising, could you please document the logic?

Reply:
Hi, sorry for the delayed answer. In MLIR this isn't computed using specific helper function at all. rather they use the count-function to reach this goal. 
`unsigned numExpectedLoops = staticTileSizes.size() - llvm::count(staticTileSizes, 0);`
As xDSL does not have any such function. I implemented this helper function myself directly. is it better to keep the logic the same as in MLIR? of coarse I can use the count function on the Array as Tuple

Reply:
That's interesting, why not replicate this logic? It seems to not use the dynamic sizes, which makes sense to me, as we can't really reason about them. The Python spelling of this function would be:
`len(static_sizes) - static_sizes.count(0)`, which is quite different from the implementation you propose.

Reply:
i see what you're saying, fixed it!